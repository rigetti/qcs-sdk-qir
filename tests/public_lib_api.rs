use std::fs::read;

use qcs::ExecutionResult;
use qcs_sdk_qir::{
    output::{self, DebugOutputFormat},
    transpile_qir_to_quil,
};

#[test]
fn transpile_qir_to_quil_bell_state() {
    let data = read("tests/fixtures/programs/bell_state.bc").unwrap();
    let output = transpile_qir_to_quil(&data).unwrap();
    insta::assert_snapshot!(output.program.to_string(true));
    insta::assert_display_snapshot!(output.shot_count);
}

#[test]
fn capture_recorded_output_and_convert() {
    let data = read("tests/fixtures/programs/record_output.bc").unwrap();
    let output = transpile_qir_to_quil(&data).unwrap();
    insta::assert_json_snapshot!(&output.recorded_output);

    let debug_format = output::try_format::<DebugOutputFormat>(
        &ExecutionResult::I8(vec![vec![1, 2], vec![2, 4], vec![3, 6]]),
        &output.recorded_output,
    )
    .unwrap();
    insta::assert_snapshot!(debug_format);
}
