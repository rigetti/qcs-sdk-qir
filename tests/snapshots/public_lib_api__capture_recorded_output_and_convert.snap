---
source: tests/public_lib_api.rs
expression: "serde_json::to_string_pretty(&output.recorded_output).unwrap()"
---
[
  "shot_start",
  {
    "result_readout_offset": 0
  },
  "tuple_start",
  {
    "result_readout_offset": 0
  },
  {
    "result_readout_offset": 1
  },
  "tuple_end",
  "array_start",
  {
    "result_readout_offset": 0
  },
  {
    "result_readout_offset": 0
  },
  "array_end",
  "shot_end"
]
